metadata-ingestion/src/datahub/ingestion/source/sql/teradata.py

```diff
@@ -1368,15 +1382,18 @@ def _create_observed_query_from_parts(
         default_database = getattr(metadata_entry, "default_database", None)
 
         # Apply Teradata-specific query transformations
-        cleaned_query = full_query_text.replace("(NOT CASESPECIFIC)", "")
+        cleaned_query = NOT_CASESPECIFIC_PATTERN.sub("", full_query_text)
 
+        # For Teradata's two-tier architecture (database.table), we should not set default_db
+        # to avoid incorrect URN generation like "dbc.database.table" instead of "database.table"
+        # The SQL parser will treat database.table references correctly without default_db
```

Comment:
A couple of questions related to this to fully understand:

* why do we call `default_database = getattr(metadata_entry, "default_database", None)` above?
* is that call what returns the `dbc` value you mention in the example?

An idea: 

In sql aggregator we handle the casing strategy for each dialect/platform 
https://github.com/datahub-project/datahub/blob/de60ca30e8d1f1c95ed672465985e575655efa28/metadata-ingestion/src/datahub/sql_parsing/sql_parsing_common.py
We may do the same for whether they are two-tier/three-tier. And so move this "logic" down to the sql aggregator layer

WDYT? It doesn't need to be done in the scope of this PR, but it could become a TODO or a task in the backlog.